todo_app/app.py

```diff
@@ -30,60 +29,156 @@ def __init__(self, id):
+    
+    @app.after_request
+    def after_request(response):
+        app.logger.info("after_request", extra={
+            "method": "{}".format(request.method),
+            "requesterIpAddr": "{}".format(request.remote_addr),
+            "url": "{}".format(request.url),
+            "status_code": "{}".format(response.status)
+        })
+        return response
+
     @login_manager.user_loader
     def load_user(user_id):        
         return UserToLogin(user_id)
     
     @login_manager.unauthorized_handler
     def unauthenticated():
-        return redirect(url_for('github_login'))
+        app.logger.info("Unauthorized attemp made.", extra={
+            "method": "{}".format(request.method),
+            "requesterIpAddr": "{}".format(request.remote_addr),
+            "url": "{}".format(request.url)
+        })
+        return redirect(url_for('login'))
     
     @app.route('/logout')
     @login_required
     def logout():
+        app.logger.info("User {} logged out of the system.".format(current_user.id), extra={
+            "method": "{}".format(request.method),
+            "requesterIpAddr": "{}".format(request.remote_addr),
+            "url": "{}".format(request.url)
+        })
         logout_user()
-        session.clear()
-        
-        github.blueprint.teardown_session
-        #return redirect(url_for('github.login'))
-        return render_template("close.html")
+        session.clear()        
+        return redirect("https://github.com/logout")
     
-    @app.route('/', methods=['GET'])
-    def github_login():
-        if not github.authorized:
-            return redirect(url_for('github.login')) 
-        account_info = github.get('/user')   
-        if account_info.ok:
-            account_info_json = account_info.json()
-            currentUserName = str(account_info_json['login'])                
+    @app.route("/login")
+    def login():
+        request_uri = client.prepare_request_uri(
+            authorization_url,
+            redirect_uri=request.base_url + "/callback",
+            scope=None,
+        )
+        return redirect(request_uri)
+
+    @app.route("/login/callback")
+    def callback():
+        code = request.args.get("code")
+        app.logger.debug("{}".format(code), extra={
+            "method": "{}".format(request.method),
+            "requesterIpAddr": "{}".format(request.remote_addr),
+            "url": "{}".format(request.url)
+        })
+
+        # Prepare and send request to get tokens! 
+        token_url, headers, body = client.prepare_token_request(
+            token_endpoint,
+            authorization_response=request.url,
+            redirect_url=request.base_url,
+            code=code,
+        )
+
+        token_response = requests.post(
+            token_url,
+            headers=headers,
+            data=body,
+            auth=(client_id, client_secret),
+        )
+
+        if token_response.status_code != 200:
```

Comment:
This could be a nice place for a log to warn something went wrong with asking for a token